pkg/infrastructure/openstack/postprovision/floatingips.go

```diff
@@ -38,7 +38,7 @@ func FloatingIPs(ctx context.Context, c client.Client, cluster *capo.OpenStackCl
 		return err
 	}
 
-	network, err := getNetworkByName(networkClient, cluster.Spec.ExternalNetworkID)
+	network, err := getNetworkByName(networkClient, cluster.Spec.ExternalNetwork.Name)
```

Comment:
I believe this is not necessary anymore, we can get the externalNetwork ID directly from `cluster.Status`
https://github.com/kubernetes-sigs/cluster-api-provider-openstack/blob/main/api/v1beta1/openstackcluster_types.go#L172-L173